password-instances/password-instances.cabal

```diff
@@ -45,27 +39,7 @@ library
       -Wall
   default-language:
       Haskell2010
-
-test-suite doctests
-  type:
-      exitcode-stdio-1.0
-  hs-source-dirs:
-      test/doctest
-  main-is:
-      doctest.hs
-  ghc-options:
-      -threaded -rtsopts -with-rtsopts=-N
-  build-depends:
-      base >=4.9 && <5
-    , base-compat
-    , doctest
-    , password
-    , password-instances
-    , QuickCheck
-    , quickcheck-instances
-    , template-haskell
-  default-language:
-      Haskell2010
+  x-docspec-extra-packages: QuickCheck quickcheck-instances password password-types
```

Comment:
Do you know how `cabal-docspec` is able to find these libraries?  Does it call Cabal under the hood to install the necessary libraries listed here?

Also, does `cabal-docspec` work when building with `stack`?

Reply:
> Do you know how cabal-docspec is able to find these libraries? Does it call Cabal under the hood to install the necessary libraries listed here?


The package must exist in the `plan.json`. It is explained [here](https://github.com/phadej/cabal-extras/blob/master/cabal-docspec/MANUAL.md)


> Also, does cabal-docspec work when building with stack?


`cabal-docspec` is looking for a `cabal-install` plan.json, I don't think `stack` provides such a thing?

Reply:
Not being able to run doctests with stack or Nix seems pretty unfortunate.  I'm not sure if I want to give that up.

I'll have to take a good look at the documentation you've linked to and think about how I feel about this.

Reply:
@cdepillabout On my personal CI I use `cabal-docspec` the same way I use `hlint`, and I don't expect the results to diverge if Nix or stack is used. If you want I can create a third set of steps that is independent of cabal / stack. 

Regarding Nix it's of course a bit more tricky since you need to patch the ELF.

Reply:
@tchoutri Ah, sorry, maybe I don't quite understand because I haven't yet read the `cabal-docspec` manual.

I was assuming that since `cabal-docspec` reads the `plan.json`, and `stack` doesn't output a `plan.json`, that people using `stack` for development can't easily run `cabal-docspec`.  (Same thing with Nix, or, for example, building `password` in Nixpkgs.)

Reply:
@cdepillabout Okay I was wrong, you indeed need to build with `cabal-install` (which is the thing that provides `plan.json`, not necessarily the `Cabal` library). 

> I don't want to give up easy support for running doctests in stack and Nix. Although, like you say on the issue, if cabal-doctest is no longer being maintained, then maybe I have no choice here...

Unfortunately as you say, this easy support might disappear one day. :/
The best path forward from here is to team up with the Cabal team so that the library provides enough features that `stack` and `cabal-install` can implement lightweight runners on top of it and reach feature parity. I don't know much about how `cabal2nix` would do it though.